01-introduction.md

```diff
@@ -0,0 +1,96 @@
+# Introduction
+
+## REST's origin story
+
+In 1989 a man by the name of [Tim Berners-Lee](https://en.wikipedia.org/wiki/Tim_Berners-Lee) 
+started working on the [HTTP](https://en.wikipedia.org/wiki/Hypertext_Transfer_Protocol) 
+protocol. HTTP is a request / response protocol, it defines a way for two computers
+to communicate with each other. This led to the world wide web as we
+know it.
+
+In the early days of the web, the HTTP protocol was not really used
+to its full extent. We mostly used [GET](https://tools.ietf.org/html/rfc7231#section-4.3.1) 
+request to retrieve pages, and the [POST](https://tools.ietf.org/html/rfc7231#section-4.3.3) 
+request to manipulate data. 
+
+This way of using POST for data manipulation is called a Remote Procedure Call 
+or [RPC](https://en.wikipedia.org/wiki/Remote_procedure_call). RPC is 
+very simple, but not very standardized, two RPC applications have
+nothing in common. In RPC designing request and responses is up mostly
+up to the individual programmer.
+
+Also some people did not think that HTTP was sufficient enough to create
+complex applications. So they invented things like [CORBA](https://en.wikipedia.org/wiki/Common_Object_Request_Broker_Architecture) 
+and [SOAP](https://en.wikipedia.org/wiki/SOAP).
+
+Then a man by the name of [Roy T. Fielding](http://roy.gbiv.com/) showed
+us the light. Turns out that there is more to HTTP than just POST and GET,
+there is also: [PUT](https://tools.ietf.org/html/rfc7231#section-4.3.4), 
+[PATCH](https://tools.ietf.org/html/rfc5789) and [DELETE](https://tools.ietf.org/html/rfc7231#section-4.3.5).
+
+Mister Fielding defined something he called REpresentational State Transfer,
+more commonly known as REST. The idea is to defined one URL for one particular
+resource and use GET, PUT, POST, PATCH and DELETE to manipulate that
+resource.
+
+REST is better than RPC because two REST applications look very similar.
```

Comment:
Do not want to nitpick but REST is better suited for web compared to SOAP and CORBA, since RPC is a very valid solution for example in online games.